runtime/fiber.c

```diff
@@ -497,7 +498,7 @@ int caml_try_realloc_stack(asize_t required_space)
   stack_used = Stack_high(old_stack) - (value*)old_stack->sp;
   wsize = Stack_high(old_stack) - Stack_base(old_stack);
   do {
-    if (wsize >= caml_max_stack_wsize) return 0;
```

Comment:
I think it would make sense to hoist the relaxed read out of the loop -- not really for performance reason (the compiler can do it anyway), but for clarity.